src/dune/findlib/findlib.ml

```diff
@@ -282,14 +283,40 @@ module Package = struct
       let known_implementations = P.Map.empty in
       let default_implementation = None in
       let wrapped = None in
+      let foreign_archives, native_archives =
+        match Path.readdir_unsorted t.dir with
+        | Error _ ->
+          (* We could add a warning like this: User_warning.emit
```

Comment:
Shouldn't this actually be an error? Can a `Package` have a non-existing/non-readable `dir`?

Reply:
This was the initial behavior for due but it turned out that it's not what we want in many cases. Sometimes, there are some permission errors and the library in question isn't used to build anything anyway. We ended up deciding to delay all errors until build time. In this case, we'll get an error anyway when we try to read any artifact of this library eventually.

Reply:
Aha, I see! I think your answer should go into the comment for the `Error _` case. This is a useful piece of information but it will get lost and forgotten very quickly.

Reply:
Done